pallets/identity/src/claims.rs

```diff
@@ -215,8 +220,15 @@ impl<T: Config> Module<T> {
         scope: Option<Scope>,
         issuer: IdentityId,
         expiry: Option<T::Moment>,
-    ) {
+    ) -> DispatchResult {
         let claim_type = claim.claim_type();
+        if let ClaimType::Custom(id) = claim_type {
+            ensure!(
+                CustomClaims::contains_key(id),
+                Error::<T>::CustomClaimTypeDoesNotExist
+            );
+        }
```

Comment:
This check is only needed in `base_add_claim`.   Also we could split that function into `base_add_claim` with this check and an `unverified_add_claim` that doesn't do the check.

Then we don't need the `unwrap`s, since those are internal calls and we know they are not adding a custom claim.